pkg/services/ngalert/state/state.go

```diff
@@ -222,7 +222,7 @@ func (a *State) Maintain(interval int64, evaluatedAt time.Time) {
 // AddErrorInformation adds annotations to the state to indicate that an error occurred.
 // If addDatasourceInfoToLabels is true, the ref_id and datasource_uid are added to the labels,
 // otherwise, they are added to the annotations.
-func (a *State) AddErrorInformation(err error, rule *models.AlertRule, addDatasourceInfoToLabels bool) {
+func (a *State) AddErrorInformation(err error, rule *models.AlertRule) {
```

Comment:
Since this method does not affect labels, then I would rather move it into compat layer and add annotations to the alert rather than the state. Not in this PR, though because we need make sure the presentation of the state in UI reads Error field.